lib/ngx/req.md

```diff
@@ -0,0 +1,146 @@
+Name
+====
+
+ngx.req - Lua API for convenience utilities for `ngx.req`.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Status](#status)
+* [Synopsis](#synopsis)
+* [Description](#description)
+* [Methods](#methods)
+    * [get_uri_ext](#get_uri_ext)
+* [Community](#community)
+    * [English Mailing List](#english-mailing-list)
+    * [Chinese Mailing List](#chinese-mailing-list)
+* [Bugs and Patches](#bugs-and-patches)
+* [Author](#author)
+* [Copyright and License](#copyright-and-license)
+* [See Also](#see-also)
+
+Status
+======
+
+This Lua module is currently considered experimental.
+
+Synopsis
+========
+
+```lua
+local ngx_req = require "ngx.req"
+
+-- GET /test.m3u8
+local ext = ngx_req.get_uri_ext()
+--> res is now "m3u8"
+```
+
+[Back to TOC](#table-of-contents)
+
+Description
+===========
+
+This Lua module provides a Lua API which implements convenience utilities for
+the `ngx.req` API.
+
+[Back to TOC](#table-of-contents)
+
+Methods
+=======
+
+All the methods of this module are static (or module-level). That is, you do
+not need an object (or instance) to call these methods.
+
+[Back to TOC](#table-of-contents)
+
+get_uri_ext
+-----------
+**syntax:** *ext = ngx_req.get_uri_ext(max?)*
+
+**context:** *set_by_lua&#42;, rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, header_filter_by_lua&#42;, body_filter_by_lua&#42;, log_by_lua&#42;, balancer_by_lua&#42;*
+
+Returns the file extension of request URL.
+
+The optional `max` argument (default 255) is a number that when specified, will return
+the `max` length of file extension when the actual length is greater then `max`.
+
+```nginx
+
+ location = /test.m3u8 {
+     content_by_lua_block {
+         local ngx_req = require "ngx.req"
+         local ext = ngx_req.get_uri_ext()
+         ngx.say(ext)
+     }
+ }
+```
+
+If the request URL has no extension the empty string "" will be returned.
+
+[Back to TOC](#table-of-contents)
+
+Community
+=========
+
+[Back to TOC](#table-of-contents)
+
+English Mailing List
+--------------------
+
+The [openresty-en](https://groups.google.com/group/openresty-en) mailing list
+is for English speakers.
+
+[Back to TOC](#table-of-contents)
+
+Chinese Mailing List
+--------------------
+
+The [openresty](https://groups.google.com/group/openresty) mailing list is for
+Chinese speakers.
+
+[Back to TOC](#table-of-contents)
+
+Bugs and Patches
+================
+
+Please report bugs or submit patches by
+
+1. creating a ticket on the [GitHub Issue Tracker](https://github.com/openresty/lua-resty-core/issues),
+1. or posting to the [OpenResty community](#community).
+
+[Back to TOC](#table-of-contents)
+
+Author
+======
+
+Thibault Charbonnier - ([@thibaultcha](https://github.com/thibaultcha))
```

Comment:
I did not write this module, the credit goes to you 😄 But feel free to put your name?